app/controllers/sessions_controller.rb

```diff
@@ -47,6 +47,12 @@ def confirm_email
     if new_user?
       user = User.create(username: session[:username], email: params[:email])
 
+      # If a new user logged in with Google, set email_for_google to the
+      # email address they used to log in.
+      if session[:email_for_google]
+        user.email_for_google = session[:email_for_google]
+      end
```

Comment:
Another style note: it's common to write such conditional assignments as one-liners (if they fit in the line length limit, not sure if in this case it does), as:
```ruby
user.email_for_google = session[:email_for_google] if user.email_for_google = session[:email_for_google].present?
```

Reply:
Looks like this one is too long (96 characters) but that pattern makes sense.